src/include/pcp/pmwebapi.h

```diff
@@ -169,6 +169,13 @@ extern int pmSeriesValues(pmSeriesSettings *, pmSeriesTimeWindow *, int, sds *,
 extern int pmSeriesQuery(pmSeriesSettings *, sds, pmSeriesFlags, void *);
 extern int pmSeriesLoad(pmSeriesSettings *, sds, pmSeriesFlags, void *);
 
+/* libpcp_web timer list interface - global, thread-safe */
+typedef void (*pmWebTimerCallBack)(void *);
+extern int pmWebTimerSetMetricRegistry(struct mmv_registry *); /* for generic server metrics */
+extern int pmWebTimerRegister(pmWebTimerCallBack, void *);
+extern int pmWebTimerRelease(int); /* deregister one timer */
+extern void pmWebTimerReleaseAll(void); /* stop and free all registered timers */
```

Comment:
On reflection I don't think we should have a pmWebTimerReleaseAll - this results in potentially sloppy programming really, and means one cannot reliably use pmWebTimerRelease (some other code can come in at any time and clobber that timer).  Each timer should be freed in the shutdown code close to the setup code for each module that registered a timer.